include/rmm/mr/device/fixed_multisize_memory_resource.hpp

```diff
@@ -86,13 +102,15 @@ class fixed_multisize_memory_resource : public device_memory_resource {
       : upstream_mr_{upstream_resource},
         size_base_{size_base},
         min_size_exponent_{min_size_exponent},
-        max_size_exponent_{max_size_exponent} {
+        max_size_exponent_{max_size_exponent},
+        min_size_bytes_{ipow(size_base, min_size_exponent)},
+        max_size_bytes_{ipow(size_base, max_size_exponent)} {
     RMM_EXPECTS(rmm::detail::is_pow2(size_base), "size_base must be a power of two");
     
     // allocate initial blocks and insert into free list
     for (std::size_t i = min_size_exponent_; i <= max_size_exponent_; i++) {
       fixed_size_mr_.emplace_back(new fixed_size_memory_resource<Upstream>(
-        upstream_resource, size_base << i, initial_blocks_per_size));
+        upstream_resource, ipow(size_base, i), initial_blocks_per_size));
```

Comment:
Was this faster than doing bit shifts?

Reply:
To use bit shifts with arbitrary base is a bit tricky.  This is not used in any inner loop.

Reply:
Ah yeah, good point about the base.